fx quant: add workflow support for torch.matmul quantization

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72444

Summary:

In https://github.com/pytorch/pytorch/pull/71783 support was added for
quantized matmul.

In this PR, the FX graph mode quantization workflow support for this
operator is added, for int8 dtypes.

Test Plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_qmatmul
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34047310](https://our.internmc.facebook.com/intern/diff/D34047310)

vkuzo committed:
fx quant: add workflow support for torch.matmul quantization

Summary:

In https://github.com/pytorch/pytorch/pull/71783 support was added for
quantized matmul.

In this PR, the FX graph mode quantization workflow support for this
operator is added, for int8 dtypes.

Test Plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_qmatmul
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34047310).

jerryzh168 reviewed (approved):
lg, looks like we need to modify some numeric suite code as well according to the ci

vkuzo committed:
Update on "fx quant: add workflow support for torch.matmul quantization"


Summary:

In https://github.com/pytorch/pytorch/pull/71783 support was added for
quantized matmul.

In this PR, the FX graph mode quantization workflow support for this
operator is added, for int8 dtypes.

Test Plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_qmatmul
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D34047310](https://our.internmc.facebook.com/intern/diff/D34047310)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34047310).
